platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/LinkedEventIntakeWiring.java

```diff
@@ -31,18 +31,21 @@
 /**
  * Wiring for the {@link LinkedEventIntake}.
  *
- * @param eventInput                  the input wire for events to be added to the hashgraph
- * @param pauseInput                  the input wire for pausing the linked event intake
- * @param consensusRoundOutput        the output wire for consensus rounds
- * @param nonAncientEventWindowOutput the output wire for the {@link NonAncientEventWindow}. This output is transformed
- *                                    from the consensus round output
- * @param flushRunnable               the runnable to flush the intake
+ * @param eventInput                        the input wire for events to be added to the hashgraph
+ * @param pauseInput                        the input wire for pausing the linked event intake
+ * @param consensusRoundOutput              the output wire for consensus rounds
+ * @param nonAncientEventWindowOutput       the output wire for the {@link NonAncientEventWindow}. This output is
+ *                                          transformed from the consensus round output
+ * @param minimumGenerationNonAncientOutput the output wire for the minimum generation non-ancient. This output is
+ *                                          transformed from the consensus round output
+ * @param flushRunnable                     the runnable to flush the intake
  */
 public record LinkedEventIntakeWiring(
         @NonNull InputWire<EventImpl> eventInput,
         @NonNull InputWire<Boolean> pauseInput,
         @NonNull OutputWire<ConsensusRound> consensusRoundOutput,
         @NonNull OutputWire<NonAncientEventWindow> nonAncientEventWindowOutput,
+        @NonNull OutputWire<Long> minimumGenerationNonAncientOutput,
```

Comment:
We should use the `nonAncientEventWindowOutput` wire to communicate this. Normally we do this sort of unpacking in the wiring layer, but for this particular situation it's better to do the unpacking within the components that care about the ancient threshold. This is part of our migration strategy for the birth round refactor. Since the nonAncientEventWindow contains both round and generation info, we won't have to rewire everything in order to throw the switch.

Reply:
We discussed this, and decided to leave it as-is to prevent merge conflicts with Cody's ongoing PCES work. He'll remove the extra wire then